ovs-transfer-obj/src/main/java/org/dcsa/ovs/transferobjects/enums/DimensionUnit.java

```diff
@@ -3,11 +3,23 @@
 import lombok.AllArgsConstructor;
 import lombok.Getter;
 
+import java.util.Objects;
+import java.util.Optional;
+
 @AllArgsConstructor
 @Getter
 public enum DimensionUnit {
   MTR("Meter"),
   FOT("Foot");
 
   private final String name;
+
+  public static Optional<DimensionUnit> valueFromString(String value) {
+    for (DimensionUnit du : DimensionUnit.values()) {
+      if (Objects.equals(value, du.toString())) {
+        return Optional.of(du);
+      }
+    }
+    return Optional.empty();
```

Comment:
Isn't this "just" a slow version `<EnumClass>.valueOf()` (with a try-catch for `IllegalArgumentException` to return `Optional.empty()`)?